anvil/checkpoint.py

```diff
@@ -23,6 +21,39 @@ def loaded_checkpoint(checkpoint):
     return cp_loaded
 
 
+def loaded_model(loaded_checkpoint, model_to_load):
+    new_model = deepcopy(
+        model_to_load
+    )  # need to copy model so we don't get weird results
+    if loaded_checkpoint is not None:
+        new_model.load_state_dict(loaded_checkpoint["model_state_dict"])
+    return new_model
+
+
+def loaded_optimizer(
+    loaded_model,
+    loaded_checkpoint,
+    optimizer,
+    optimizer_params,
+):
+    instance = optimizer(loaded_model.parameters(), **optimizer_params)
+    if loaded_checkpoint is not None:
+        instance.load_state_dict(loaded_checkpoint["optimizer_state_dict"])
+    return instance
```

Comment:
I think it's nicer to load the optimizer and scheduler here, alongside the model, and just have one action that does it rather than one per possible optimizer.